arangod/VocBase/Methods/Collections.cpp

```diff
@@ -1116,19 +1116,31 @@ arangodb::Result Collections::checksum(LogicalCollection& collection,
   return trx.finish(res);
 }
 
+/// @brief the list of collection attributes that are allowed by user-input
+/// this is to avoid retyping the same list twice
+#define COMMON_ALLOWED_COLLECTION_INPUT_ATTRIBUTES                            \
+  StaticStrings::DataSourceSystem, StaticStrings::DataSourceId, "keyOptions", \
+      StaticStrings::WaitForSyncString, StaticStrings::CacheEnabled,          \
+      StaticStrings::ShardKeys, StaticStrings::NumberOfShards,                \
+      StaticStrings::DistributeShardsLike, "avoidServers",                    \
+      StaticStrings::IsSmart, StaticStrings::ShardingStrategy,                \
+      StaticStrings::GraphSmartGraphAttribute, StaticStrings::Schema,         \
+      StaticStrings::SmartJoinAttribute, StaticStrings::ReplicationFactor,    \
+      StaticStrings::MinReplicationFactor, /* deprecated */                   \
+      StaticStrings::WriteConcern, "servers"
+
 arangodb::velocypack::Builder Collections::filterInput(
-    arangodb::velocypack::Slice properties) {
+    arangodb::velocypack::Slice properties, bool allowDC2DCAttributes) {
+#ifdef USE_ENTERPRISE
+  if (allowDC2DCAttributes) {
+    return velocypack::Collection::keep(
+        properties,
+        std::unordered_set<std::string>{
+            StaticStrings::IsDisjoint, StaticStrings::InternalValidatorTypes,
```

Comment:
Why isn't the `isDisjoint` attribute kept without DC2DC? Isn't this a necessary flag when creating a collection inside a disjoint SmartGraph?

Reply:
Correct, but from a user-perspective this needs to be created over the GraphsAPI to have Vertex and Edge collections aligned.
This is a dangerous attribute and if the definitions do not match the graphs will not work as expected.
The GraphsAPI is already past this filtering of attributes.